### PR TITLE
EEPROM & Raman Intensity Correction fixes

### PR DESCRIPTION
**Pull request by mzieg**
Changes:

- fix writing EEPROM on ARM
- fixed EEPROM write return codes
- fixed and documented Raman Intensity Correction

### CODE REVIEW

File: WasatchVCPPLib/WasatchVCPPLib/WasatchVCPPWrapper.cpp [hunk 1]
```diff
@@ -765,57 +764,46 @@ int wp_get_raman_intensity_factors(int specIndex, double* factors, int factorsLe
...
-    float scaled;
-    float expanded;
-    int i;
-    int j;
+    if (eeprom.intensityCorrectionCoeffs.empty())
+        return WP_ERROR_NO_CALIBRATION;
     
     // expanding coeffs and calculating factors
-    if (!eeprom.intensityCorrectionCoeffs.empty())
+    for (int i = 0; i < factorsLen; i++)
```
Comment (samiebee43) [inline WasatchVCPPLib/WasatchVCPPLib/WasatchVCPPWrapper.cpp:773]: Keep iteration logic in for-loop.
```
int ROIHorizEnd = eeprom.ROIHorizStart + factorsLen;
for (int i = eeprom.ROIHorizStart; i < ROIHorizEnd; i++)
```

You could rename `i` to `px`, but we don't need both variables. If you rename you should change `x_to_i` to `x_to_px`

Comment (mzieg) [reply WasatchVCPPLib/WasatchVCPPLib/WasatchVCPPWrapper.cpp:773]: If we did that, then we'd be doing this in the end, which doesn't feel like an improvement. Having to constantly juggle offsets without clearly labeled indexes is what led to the original code being buggy.
```
factors[i - eeprom.ROIHorizStart] = pow(10, logTen);
```

Comment (samiebee43) [reply WasatchVCPPLib/WasatchVCPPLib/WasatchVCPPWrapper.cpp:773]: You're right I didn't see Ln783

File: demo-linux/demo.cpp [hunk 2]
```diff
@@ -130,12 +130,12 @@ void writeToEEPROM()
...
     printf("Original contents of EEPROM page %d: [%s]\n", USER_DATA, orig.c_str());
 
-    // generate new page contents
+    // generate new page contents 
     string writeString = "Wasatch.VCPP demo at " + timestamp();
 
     // write the EEPROM page
     printf("Overwriting page contents with: [%s]\n", writeString.c_str());
-    strncpy((char*)buf, writeString.c_str(), writeString.length());
+    strncpy((char*)buf, writeString.c_str(), PAGE_SIZE);
```
Comment (samiebee43) [inline demo-linux/demo.cpp:138]: Yes this prevents buffer overrun, but it is still vulnerable to non-termination. Use `PAGE_SIZE-1` to guarantee buff has a null terminator.

Comment (mzieg) [reply demo-linux/demo.cpp:138]: null-terminators are not required in fixed-width fields

**Pull request merged**